src/main/java/com/mkul/game/ox/boardcheck/BoardCheckFirstDiagonal.java

```diff
@@ -0,0 +1,29 @@
+package com.mkul.game.ox.boardcheck;
+
+import com.mkul.game.ox.BoardScore;
+import com.mkul.game.ox.Field;
+import com.mkul.game.ox.FieldValue;
+
+class BoardCheckFirstDiagonal implements BoardCheck {
+
+    /**
+     * {@inheritDoc}
+     * check wining on first Diagonal
+     */
+    @Override
+    public boolean check(BoardScore boardScore, int winingSize, Field fieldToCheck, FieldValue fieldValue) {
+        int steps = 1;
+        int score = 1;
+        while (winingSize != steps) {
+            if (boardScore.isFieldExist(fieldToCheck.nextRight(steps).nextUp(steps)) && boardScore.getValue(fieldToCheck.nextRight(steps).nextUp(steps)) == fieldValue) {
+                score++;
+            }
+
+            if (boardScore.isFieldExist(fieldToCheck.nextLeft(steps).nextDown(steps)) && boardScore.getValue(fieldToCheck.nextLeft(steps).nextDown(steps)) == fieldValue) {
```

Comment:
To możliwe, że przekracza dopuszczalną długość. Daj na wszystkie klasy autoreformat (`CTRL+ALT+L`).

Reply:
Pewnie zależnie od ustawionego stylu, dał bym enter przed `&&`.